### PR TITLE
Randomize (and limit) mining jobs

### PR DESCRIPTION
**Pull request by killerwhile**
This MR is a test to make this mining pool ASIC compatible. The first change is to randomize the mining jobs sent to the miners, because there are some miners which only mine the first job returned, and this one is at the very moment always on chain 0->0.

### CODE REVIEW

File: lib/jobManager.js [hunk 1]
```diff
@@ -79,8 +79,13 @@ function JobManager(jobExpiryPeriod){
             job.jobId = jobId;
             return new blockTemplate(job, now);
         })
-        _this.validJobs.addJobs(miningJobs, now);
-        _this.emit('newJobs', miningJobs);
+        // randomize and limit miningJobs
+        randomizedMiningJobs = miningJobs.map(value => ({ value, sort: Math.random() }))
```
Comment (nnslvp) [inline lib/jobManager.js:83]:
```suggestion
        var randomizedMiningJobs = miningJobs.map(value => ({ value, sort: Math.random() }))
```

File: lib/jobManager.js [hunk 2]
```diff
@@ -79,8 +79,13 @@ function JobManager(jobExpiryPeriod){
...
+            .sort((a, b) => a.sort - b.sort)
+            .map(({ value }) => value)
+            .slice(0,3)
```
Comment (nnslvp) [inline lib/jobManager.js:86]:
```suggestion
            .slice(0, 3)
```

Comment (nnslvp) [inline lib/jobManager.js:86]: How exactly was the limit of 3 chosen ?

**Comment by polarker** [comment]
This feature has been implemented and merged. Please check the merged PR here: https://github.com/alephium/mining-pool/pull/67

**Pull request closed without merging**